src/sentry/api/endpoints/project_event_details.py

```diff
@@ -41,19 +42,52 @@ def get(self, request, project, event_id):
         :auth: required
         """
 
+        use_snuba = request.GET.get('enable_snuba') == '1' and \
+            options.get('snuba.events-queries.enabled')
+
+        if not use_snuba:
+            return self.get_legacy(request, project, event_id)
+
+        # We might get a database id instead of an event id
+        if not is_event_id(event_id):
+            event = Event.objects.from_event_id(event_id, project.id)
+
+            if event is None:
+                return Response({'detail': 'Event not found'}, status=404)
+
+            event_id = event.event_id
```

Comment:
Is it possible to put this block in some kind of utility function in EventManager? 

Basically something in line with this comment.
https://github.com/getsentry/sentry/blob/master/src/sentry/db/models/manager.py#L355-L356

Basically we should try and make sure everywhere in the code is using `event_id` and if it isn't we can resolve `id` to `event_id` for now (and maybe log the times that this happens so we can fix them?)